go/vt/vtgate/evalengine/compiler_asm.go

```diff
@@ -3795,6 +3795,22 @@ func (asm *assembler) Fn_LAST_DAY() {
 	}, "FN LAST_DAY DATETIME(SP-1)")
 }
 
+func (asm *assembler) Fn_TO_DAYS() {
+	asm.emit(func(env *ExpressionEnv) int {
+		if env.vm.stack[env.vm.sp-1] == nil {
+			return 1
+		}
```

Comment:
@vmg It's actually not since `c.asm.Convert_xD` can result in a `NULL` in the path when it's not already a `Date` instance. 

That's something that you can see in more places for date conversions. We could also have `c.asm.Convert_xD` return a jump in case of nil and add that, but that's a separate refactor / change that I don't think we should make here.

Reply:
:+1: